src/persist-client/src/internal/machine.rs

```diff
@@ -166,15 +166,25 @@ where
         reader_id: &ReaderId,
         lease_duration: Duration,
         heartbeat_timestamp_ms: u64,
-    ) -> (Upper<T>, ReaderState<T>) {
+    ) -> ReaderState<T> {
         let metrics = Arc::clone(&self.metrics);
-        let (seqno, (shard_upper, read_cap), _maintenance) = self
+        let (_seqno, reader_state, _maintenance) = self
             .apply_unbatched_idempotent_cmd(&metrics.cmds.register, |seqno, state| {
                 state.register_reader(reader_id, seqno, lease_duration, heartbeat_timestamp_ms)
             })
             .await;
-        debug_assert_eq!(seqno, read_cap.seqno);
-        (shard_upper, read_cap)
+        // Usually, the reader gets an initial seqno hold of the seqno at which
+        // it was registered. However, on a tombstone shard the seqno hold
+        // happens to get computed as the tombstone seqno + 1. The real
+        // invariant we want to protect here is that the hold is >= the
+        // seqno_since, so validate that instead of anything more specific.
```

Comment:
I feel silly but I've been staring at this for a bit and don't know why it's tombstone seqno + 1

Reply:
added a hopefully clarifying parenthetical to the comment, lemme know if that helps